### ISSUE TITLE
SqlDataProvider has error at code time, but compiles successfully

### ISSUE THREAD

**Issue by tylerhartwig** [fsprojects/SQLProvider#528]
### Description

At Design or Code time, while creating the SqlDataProvider type an error returned in the IDE.

![ide error](https://user-images.githubusercontent.com/9355929/37577264-384100a4-2aff-11e8-9a96-6dbf5d9e28da.png)

This IDE error was in Jetbrains Rider. I experienced a similar, but less helpful error in Visual Studio for Mac. It was along the lines of "Failure has occurred while loading a type."

### Repro steps
```

open System
open System.Collections.Generic
open System.IO
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore
open Microsoft.AspNetCore.Hosting
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.Logging

open FSharp.Data.Sql

[<Literal>]
let connStr = "User ID=test;Host=localhost;Port=5432;Database=test;Password=test"

[<Literal>]
let resolutionPath = "libraries"

type sql = SqlDataProvider<Common.DatabaseProviderTypes.POSTGRESQL, connStr, ResolutionPath = resolutionPath, Owner = "schema">

[<EntryPoint>]
let main argv =
    let ctx = sql.GetDataContext()
    let test1 = 
        query {
            for t in ctx.Schema.Test do
            select t.Test
        } |> Seq.head
        
    printfn "Hello %c!" test1
    0
```

Is my program file, and my fsproj is as follows:

```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>
  <PropertyGroup>
    <FscToolPath>/Library/Frameworks/Mono.framework/Versions/Current/Commands</FscToolPath>
    <FscToolExe>fsharpc</FscToolExe>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <DebugSymbols>false</DebugSymbols>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Controllers/ValuesController.fs" />
    <Compile Include="Startup.fs" />
    <Compile Include="Program.fs" />
  </ItemGroup>
  <!-- .NET 4.61 is Net Standard 2.0 compatible so use what ever libraries.
  However non-Windows typeprovider will call Mono on compile-time. Mono is not Net Standard 2.0 compatible.
  So Mono will use Net 4.5.1 libraries. Typeprovider handles that on compilation, and after compile your 
  assembly will ve .NET Core compatible. -->
  <ItemGroup>
    <!-- Mono fix -->
    <Reference Condition="'$(IsWindows)' != 'true' Or '$(TargetFramework)' == 'netstandard2.0'" Include="FSharp.Data.SqlProvider">
      <HintPath>../.nuget/packages/sqlprovider/1.1.37/lib/net451/FSharp.Data.SqlProvider.dll</HintPath>
      <!--<HintPath>../../../bin/net451/FSharp.Data.SqlProvider.dll</HintPath>-->
      <!--HintPath>../../packages/SQLProvider/lib/net451/FSharp.Data.SqlProvider.dll</HintPath-->
    </Reference>
  </ItemGroup>
  <!--<ItemGroup>-->
    <!--<Reference Include="System.Threading.Tasks.Extensions">-->
      <!--<HintPath>../.nuget/packages/system.threading.tasks.extensions/4.4.0/lib/portable-net45+win8+wp8+wpa81/System.Threading.Tasks.Extensions.dll</HintPath>-->
    <!--</Reference>-->
  <!--</ItemGroup>-->
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.0" />
    <PackageReference Include="Npgsql" Version="3.2.5" />
    <!--<PackageReference Include="SQLProvider" Version="1.1.37" />-->
    <PackageReference Include="System.Console" Version="4.3.0" />
    <PackageReference Include="System.Data.Common" Version="4.3.0" />
    <PackageReference Include="System.Reflection" Version="4.3.0" />
    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.4.0" />
    <PackageReference Include="System.Runtime" Version="4.3.0" />
    <PackageReference Include="System.Runtime.Extensions" Version="4.3.0" />
    <PackageReference Include="System.Runtime.Serialization.Formatters" Version="4.3.0" />
    <PackageReference Include="System.Threading.Tasks.Extensions" Version="4.4.0" />
  </ItemGroup>
</Project>
```

### Expected behavior

No error should be provided. 

### Actual behavior

`type sql = SqlDataProvider<Common.DatabaseProviderTypes.POSTGRESQL, connStr, ResolutionPath = resolutionPath, Owner = "schema">`

The mentioned line gives red squiggles, and subsequent lines provide no intellisense. 

Everything compiles and runs successfully however. 

### Related information 

* Npgsql
* .NET Core
* MacOS 

Any help to fix this would be very helpful, as the intellisense is one of the best parts of TypeProviders. 

Thanks

**Comment by Thorium** [comment]
`System.Transactions` should be part of .NET Core 2.0 and not an external package, so it should work... So I expect this is more of .NET Core problem than SQLProvider. You don't use transactions directly, but GetDataContext() -method has on override that has transactions so it could mess the editors somehow.

Maybe deleting the bin and obj and rerun `dotnet restore` and `dotnet build` could help...

**Comment by tylerhartwig** [comment]
I tried deleting the `bin` and `obj` folders tonight with no luck. As I mentioned, everything builds and runs perfectly, just an IDE error unfortunately. 

Visual Studio for Mac decided it liked me tonight however, and gave a helpful error in the IDE.

Screenshot: 
![screen shot 2018-03-19 at 10 07 06 pm](https://user-images.githubusercontent.com/9355929/37633863-13245d52-2bc2-11e8-949d-26e5d3451039.png)

Does this help at all? my theory is the IDE knows it's a .NET Core project and is trying to compile and run the tooling using the wrong dlls. Not sure if there is a way to remedy that, or if I'm wrong about that. 

Thanks for the help so far :)

**Comment by Thorium** [comment]
Hmm. Maybe the editors cannot figure out the relative resolutionPath.

Try to use absolute, or with __SOURCE_DIRECTORY__ , like:
```fsharp
let [<Literal>] resolutionPath = __SOURCE_DIRECTORY__ + @"/../libraries" 
```

**Comment by tylerhartwig** [comment]
That's good thinking.

I tried it this morning, did not change any of the behavior, however the tool tip is returning a slightly different error in Rider now.

![screen shot 2018-03-20 at 7 43 23 am](https://user-images.githubusercontent.com/9355929/37655164-82b2233e-2c12-11e8-80cc-75ff52cf7aab.png)

And Visual Studio for Mac decided to not be helpful this go-around (just gives a generic error about not being able to load types).  

Do I need to also copy a dll to support `System.Transactions` into the `libraries` path? I tried to do this, however there is no nuget package for it. I may try copying that from the runtime tonight if I don't come up with any other solutions.

**Comment by tylerhartwig** [comment]
Update:

I copied `System.Transactions.dll` into the `libraries` folder, and Rider did not work any better, however I opened Visual Studio and it started working. I proceeded to remove it from the folder, and Visual Studio continued to work. 

I'm guessing the copying of the dll didn't do anything for Visual Studio, but rather Visual Studio worked this time around (I've experienced other things like this with Visual Studio for Mac in general) 

The remaining issue must be something exclusive to Rider then. Thank you all the help, the absolute path seemed to help the problem.
